### ISSUE TITLE
Optimize pytorch run as good as possible. 

### ISSUE THREAD

**Issue by vegovs** [gil-uav/semantic-image-segmentation#1]
- [x] Asynchronous data loading and augmentation enabled.
- [x] cuDNN autotuner enabled.
- [x] Automatic mixed precision, i.e use of float16 in tensorcores.
- [ ] Increase batch-size due to reduced memory usage.
    - [ ] Tune learning rate.
    - [ ] Add learning rate warmup.
    - [x] Add learning decay.
    - [ ] Add weight decay.
    - [ ] Use optimizer for large-barch training(LARS, LAMB, NVLAMB, NovoGrad).
- [x] Disable bias for convolutions directly followed by a batch norm.
- [x] Use parameter.grad = None.
- [x] Disbale debut APIs in production-mode.
- [ ] Opt for [DistributedDataParallel](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html)
instead of [DataParallel](https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html) to improve Multi-GPU efficiency.
    - [ ] Load-balance workload on multiple GPUs.
- [ ] Use fused building blocs from [APEX](https://github.com/NVIDIA/apex).
- [ ] Checkpoint to recompute intermediates.
- [ ] Fuse pointwise operations.
